### DISCUSSION TITLE
[과목 2] 제 3장 관리 구문

### DISCUSSION THREAD

**Discussion by white-gyu** [flataex/sqlp-study#16] [연습문제]
### 연습 문제

**문제 1. 데이터를 입력하기 위해 사용하는 SQL 명령어는 무엇인가?**

① CREATE ② UPDATE ③ INSERT ④ ALTER

**문제 2. 데이터의 입력 및 수정 작업을 한 번에 할 수 있는 SQL 명령어는 무엇인가?**

① MERGE ② MODIFY ③ TRUNCATE ④ DELETE

**문제 3. Commit과 Rollback의 장점으로 적합하지 않은 것은 무엇인가?**

① 데이터 무결성을 보장한다.
② 영구적인 변경을 하기 전에 데이터의 변경 사항을 확인할 수 있다. 
③ 영구적인 변경을 할 수 없게 한다.
④ 논리적으로 연관된 작업을 그룹핑해 처리할 수 있다.

**문제 4. 다음 중 테이블 명으로 가능한 것은 무엇인가?**

① EMP100 ② 100EMP ③ EMP-100 ④ 100-EMP

**문제 5. 다음 중 제약조건의 종류로 적절하지 않은 것은 무엇인가?**

① CHECK ② UNION ③ PRIMARY KEY ④ NOT NULL

**문제 6. TRUNCATE TABLE 문에 대한 설명으로 적절한 것은 무엇인가?**

① COMMIT 명령어를 수행해야 영구적으로 반영된다.
② 내부 처리 방식이나 Auto Commit 특성 등으로 인해 DML로 분류한다. 
③ 전체 데이터를 삭제하는 경우, DELETE 문에 비해 훨씬 더 느리다.
④ 테이블 구조는 그대로 유지한 채 데이터만 전부 삭제하는 명령어다.

**문제 7. 다음 중 사용자로부터 데이터베이스 권한을 회수하는 명령어는 무엇인가?**

① ROLLBACK ② REVOKE ③ DROP ④ GRANT

**Comment by white-gyu** [comment]
### 연습 문제

**문제 1. 데이터를 입력하기 위해 사용하는 SQL 명령어는 무엇인가?**

① CREATE ② UPDATE ③ INSERT ④ ALTER

> 3

**문제 2. 데이터의 입력 및 수정 작업을 한 번에 할 수 있는 SQL 명령어는 무엇인가?**

① MERGE ② MODIFY ③ TRUNCATE ④ DELETE

> 1

**문제 3. Commit과 Rollback의 장점으로 적합하지 않은 것은 무엇인가?**

① 데이터 무결성을 보장한다.
② 영구적인 변경을 하기 전에 데이터의 변경 사항을 확인할 수 있다. 
③ 영구적인 변경을 할 수 없게 한다.
④ 논리적으로 연관된 작업을 그룹핑해 처리할 수 있다.

> 3

**문제 4. 다음 중 테이블 명으로 가능한 것은 무엇인가?**

① EMP100 ② 100EMP ③ EMP-100 ④ 100-EMP

> 1

**문제 5. 다음 중 제약조건의 종류로 적절하지 않은 것은 무엇인가?**

① CHECK ② UNION ③ PRIMARY KEY ④ NOT NULL

> 2

**문제 6. TRUNCATE TABLE 문에 대한 설명으로 적절한 것은 무엇인가?**

① COMMIT 명령어를 수행해야 영구적으로 반영된다.
② 내부 처리 방식이나 Auto Commit 특성 등으로 인해 DML로 분류한다. 
③ 전체 데이터를 삭제하는 경우, DELETE 문에 비해 훨씬 더 느리다.
④ 테이블 구조는 그대로 유지한 채 데이터만 전부 삭제하는 명령어다.

> 4
> COMMIT 명령어를 수행하지 않더라도 영구적으로 반영되며, 로그가 없으므로 복구가 어렵다.
내부 처리 방식이나 Auto Commit 특성 등으로 인해 "DDL"로 분류한다.
전체 데이터를 삭제하는 경우 DELETE 문에 비해 훨씬 더 빠르다.

**문제 7. 다음 중 사용자로부터 데이터베이스 권한을 회수하는 명령어는 무엇인가?**

① ROLLBACK ② REVOKE ③ DROP ④ GRANT

> 2

**Comment by Psjepark** [comment]
> ### 연습 문제
> **문제 1. 데이터를 입력하기 위해 사용하는 SQL 명령어는 무엇인가?**
> 
> ① CREATE ② UPDATE ③ INSERT ④ ALTER
>  3
> **문제 2. 데이터의 입력 및 수정 작업을 한 번에 할 수 있는 SQL 명령어는 무엇인가?**
> 
> ① MERGE ② MODIFY ③ TRUNCATE ④ DELETE
> 1
> **문제 3. Commit과 Rollback의 장점으로 적합하지 않은 것은 무엇인가?**
> 
> ① 데이터 무결성을 보장한다. ② 영구적인 변경을 하기 전에 데이터의 변경 사항을 확인할 수 있다. ③ 영구적인 변경을 할 수 없게 한다. ④ 논리적으로 연관된 작업을 그룹핑해 처리할 수 있다.
> 3
> **문제 4. 다음 중 테이블 명으로 가능한 것은 무엇인가?**
> 
> ① EMP100 ② 100EMP ③ EMP-100 ④ 100-EMP
> 1
> **문제 5. 다음 중 제약조건의 종류로 적절하지 않은 것은 무엇인가?**
> 
> ① CHECK ② UNION ③ PRIMARY KEY ④ NOT NULL
> 2
> **문제 6. TRUNCATE TABLE 문에 대한 설명으로 적절한 것은 무엇인가?**
> 
> ① COMMIT 명령어를 수행해야 영구적으로 반영된다. ② 내부 처리 방식이나 Auto Commit 특성 등으로 인해 DML로 분류한다. ③ 전체 데이터를 삭제하는 경우, DELETE 문에 비해 훨씬 더 느리다. ④ 테이블 구조는 그대로 유지한 채 데이터만 전부 삭제하는 명령어다.
> 4
> **문제 7. 다음 중 사용자로부터 데이터베이스 권한을 회수하는 명령어는 무엇인가?**
> 
> ① ROLLBACK ② REVOKE ③ DROP ④ GRANT
2
